pkg/authentication/scram/scram.go

```diff
@@ -28,37 +27,108 @@ const (
 	sha256StoredKeyKey = "sha-256-stored-key"
 )
 
-// EnsureScram will configure all of the required Kubernetes resources for SCRAM-SHA to be enabled.
+// Configurable is an interface which any resource which can configure ScramSha authentication should implement.
+type Configurable interface {
+	// GetScramOptions returns a set of Options which can be used for fine grained configuration.
+	GetScramOptions() Options
+
+	// GetScramUsers returns a list of users which will be mapped to users in the AutomationConfig.
+	GetScramUsers() []User
+
+	// GetAgentScramCredentialsNamespacedName returns the NamespacedName of the secret which will store the generated credentials for the agent.
+	GetAgentScramCredentialsNamespacedName() types.NamespacedName
+
+	// NamespacedName returns the NamespacedName for the resource that is being configured.
+	NamespacedName() types.NamespacedName
+}
+
+// Role is a struct which will map to automationconfig.Role.
+type Role struct {
+	// Name is the name of the role.
+	Name string
+
+	// Database is the database this role applies to.
+	Database string
+}
+
+// User is an interface which must be implemented by a type which will map to an
+// automationconfig.MongoDBUser.
+type User struct {
```

Comment:
contains information for the User that will be in the automation config and values related to the password secret